app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/splash/SplashScreen.kt

```diff
@@ -1,53 +1,202 @@
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.*
+import androidx.compose.runtime.*
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalDensity
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.input.PasswordVisualTransformation
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.IntOffset
 import androidx.compose.ui.unit.dp
-import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
 import com.kks.nimblesurveyjetpackcompose.R
+import com.kks.nimblesurveyjetpackcompose.ui.theme.Concord
+import com.kks.nimblesurveyjetpackcompose.ui.theme.White18
 import com.kks.nimblesurveyjetpackcompose.viewmodel.splash.SplashViewModel
 import com.ramcosta.composedestinations.annotation.Destination
+import com.ramcosta.composedestinations.annotation.NavHostParam
+import com.ramcosta.composedestinations.annotation.RootNavGraph
+import kotlinx.coroutines.launch
 
-@Destination(start = true)
+@RootNavGraph(start = true)
+@Destination
 @Composable
-fun SplashScreen(viewModel: SplashViewModel = viewModel()) {
+fun SplashScreen(
+    @NavHostParam splashTime: Long = 2000L,
+    viewModel: SplashViewModel = hiltViewModel()
+) {
+    val showLoginComponents by viewModel.shouldNavigateToLogin.collectAsState()
+    val logoOffset = remember { Animatable(Offset(0f, 0f), Offset.VectorConverter) }
+    val positionToAnimate = -LocalDensity.current.run { 221.dp.toPx() }
+
     LaunchedEffect(
-        key1 = viewModel.shouldNavigateToLogin.value
+        key1 = showLoginComponents
     ) {
-        if (viewModel.shouldNavigateToLogin.value) {
-            // TODO: Navigate to Login
+        if (showLoginComponents) {
+            launch {
+                logoOffset.animateTo(
+                    Offset(logoOffset.value.x, positionToAnimate),
+                    spring(
+                        dampingRatio = Spring.DampingRatioLowBouncy,
+                        stiffness = StiffnessVeryLow
+                    )
+                )
+            }
         }
     }
-
     Box(
         modifier = Modifier.fillMaxSize(),
         contentAlignment = Alignment.Center
     ) {
         Image(
```

Comment:
Why don't we wrap this `Image` in a `Crossfade` so it transitions cleanly between blurred and non-blurred states? 😄 
Something like:
```kotlin
Crossfade(
    targetState = showLoginComponents,
    animationSpec = tween(1000)
) { isBlurred -> {
    Image(
        painter = painterResource(id = if (isBlurred) R.drawable.ic_overlay else R.drawable.splash_bg),
        ...
    )
}